### PR TITLE
Task dev 7

### PR DESCRIPTION
**Pull request by SiarheySkuratovich**

### CODE REVIEW

File: SideReading.java [hunk 1]
```diff
@@ -0,0 +1,21 @@
+import java.util.Scanner;
+
+/**
+ * Gets sides from command line and writes to array.
+ */
+public class SideReading {
+
+  /**
+   * Gets sides from command line and writes to array.
```
Comment (AlexanderSoroka) [inline SideReading.java]: Not from command line - from console

File: Triangle.java [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+   * An array of sides values.
+   */
+  double[] sides;
+
+  /*
+   * Creates the Triangle object using (@link SideReading#readSide(double[]).
+   */
+  Triangle() {
+    sides = new double[3];
+    SideReading getting = new SideReading();
```
Comment (AlexanderSoroka) [inline Triangle.java]: Single responsibility violation, class Triangle must not read sides in any way - pass sides as constructor arguments

File: Triangle.java [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+    getting.readSide(sides);
+  }
+
+  /*
+   * Cheks on the type of triangle.
+   */
+  public void check() {
+    if(sides[0] == sides[1] && sides[1] == sides[2]) {
+      System.out.println("Triangle is equilateral");
+    } else if (sides[0] == sides[1] || sides[1] == sides[2] || sides[0] == sides[2]) {
```
Comment (AlexanderSoroka) [inline Triangle.java]: All these checks must be implemented in separate classes

**Review by AlexanderSoroka — CHANGES_REQUESTED** [review]
There is must be class fro each triangle type

File: Triangle.java (hunk 3 above)
Comment (SiarheySkuratovich) [reply Triangle.java]: @AlexanderSoroka exactly classes or i can leave as methods of Triangle class?

File: SideReading.java [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+
+/**
+ * Gets sides of geometric shapes.
+ */
+public class SideReading {
+
+  /**
+   * Gets trangle sides from console and returns them as array.
+   */
+  public double[] readSide() {
```
Comment (AlexanderSoroka) [inline SideReading.java]: Why readSide (one side) but return array (many sides)?

File: SideReading.java [hunk 5]
```diff
@@ -0,0 +1,22 @@
+import java.util.Scanner;
+
+/**
+ * Gets sides of geometric shapes.
+ */
+public class SideReading {
+
+  /**
+   * Gets trangle sides from console and returns them as array.
```
Comment (AlexanderSoroka) [inline SideReading.java]: javadoc violation

File: Triangle.java [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+   * An array of sides values.
+   */
+  double[] sides;
+
+  /*
+   * Creates the Triangle object.
+   * @param gottenSides we get from console. 
+   */
+  Triangle(double[] gottenSides) {
+    sides = gottenSides;
```
Comment (AlexanderSoroka) [inline Triangle.java]: Missed validation

**Review by AlexanderSoroka — CHANGES_REQUESTED** [review]
For each triangle type you must have separate

File: Triangle.java [hunk 7]
```diff
@@ -0,0 +1,48 @@
...
+      sides = gottenSides;
+    }
+  }
+
+  /*
+   * Checks whether triangle is equilateral.
+   * @return true if the triangle is Equilateral and false if it is not.
+   */
+  public boolean checkForEquilateral() {
+    if(sides[0] == sides[1] && sides[1] == sides[2]) {
```
Comment (AlexanderSoroka) [inline Triangle.java]: Return boolean value instead using if

**Review by AlexanderSoroka — CHANGES_REQUESTED** [review]
For each triangle type you must have separate
Hide outdated

**Pull request closed without merging**